src/MuonsProducer.cc

```diff
@@ -16,12 +18,25 @@ void MuonsProducer::produce(edm::Event& event, const edm::EventSetup& eventSetup
     const reco::Vertex& primary_vertex = (*vertices_handle)[0];
 
     double rho = *rho_handle;
-
+    rochcor2015 *rmcor = new rochcor2015();
```

Comment:
You have a memory leak here. Move this as a class member, something like
`rochcor2015 m_rc_corrector;`

No need for it to be a pointer.